py/objtype.c

```diff
@@ -1011,6 +1011,31 @@ STATIC void type_attr(mp_obj_t self_in, qstr attr, mp_obj_t *dest) {
             dest[0] = MP_OBJ_NEW_QSTR(self->name);
             return;
         }
+        if (attr == MP_QSTR___bases__) {
+            const void * parent = self->parent;
+            if (parent == MP_OBJ_NULL) {
+                mp_obj_t tuple[1] = { MP_OBJ_FROM_PTR(&mp_type_object) };
+                dest[0] = mp_obj_new_tuple(1, tuple);
+                return;
+            } else {
+                mp_obj_t parent_obj = MP_OBJ_FROM_PTR(self->parent);
+                #if MICROPY_MULTIPLE_INHERITANCE
+                if (mp_obj_is_type(parent_obj, &mp_type_tuple)) {
+                    dest[0] = parent_obj;
+                    return;
+                } else
+                #endif
+                if (mp_obj_is_type(parent_obj, &mp_type_type)) {
+                    mp_obj_t tuple[1] = { parent_obj };
+                    dest[0] = mp_obj_new_tuple(1, tuple);
+                    return;
+                } else {
+                    mp_raise_TypeError(NULL);
+                }
+
+            }
+            mp_raise_NotImplementedError("__bases__ just exploded.");
```

Comment:
Don't think you need this line :)

Reply:
Every branch must have an end, I just wasn't sure if we had ended yet, I'll clean that up.